design.md

```diff
@@ -0,0 +1,75 @@
+* `IsRunningState()` - true if the process is starting or running.
+* `IsFinalState()` - true if the process is in a final state
+
+## GRPC API
+Please refer to the service proto file for the full definition of the grpc api
+* `rpc Register (JobRequest) returns (JobStatus) {}` - Registers a job without starting it. Returns a unique job id for subsequent use with the other api methods. A status `Initial` is returned to indicate the job has been added.
+* `rpc Start (StartRequest) returns (stream JobStatus) {}` - Start a job that was previously registered using the provided job id. Returns a stream of `JobStatus` for the client to listen to status updates. Returns a `NOT_FOUND` error if the provided job id doesn't exist in the system. It is not an error to call `Start(...)` for a running job or a job in a final state. The stream is closed when the job reaches a final state.
+    * Calling `Start(...)` when the process is running simply returns the job status stream.
+    * Calling `Start(...)` when the process is in a final state returns the final status and closes the stream.
+* `rpc StatusStream (StatusStreamRequest) returns (stream JobStatus) {}` - `StatusStream(...)` is a convenience method that returns a stream for the client to listen to job status updates using the provided job id. Returns a `NOT_FOUND` error if the provided job id doesn't exist in the system. It works similary to `Start(...)` except it does not start the job if it is not running. A status of `Initial` is returned if the job wasn't started. 
+* `rpc Status(StatusRequest) returns (JobStatus) {}` - `Status(...)` simply returns the current status of a job. Returns a `NOT_FOUND` error if the job doesn't exist in the system. It is recommended to use `Start(...)` or `StatusStream(...)` to get job status updates instead of polling `Status(...)` repeatedly.
+* `rpc Stop(StopRequest) returns (JobStatus) {}` - Stops a running job with the provided job id and returns the final status. Returns a `NOT_FOUND` error if the provided job id doesn't exist. The api call does nothing if the job is not running. A status `Initial` is returned in this case.
+* `rpc GetOuput(GetOutputRequest) returns (stream JobOutput) {}` - Returns the job output stream given the job id. A `NOT_FOUND` error is returned if the provided job id does not exist. 
+    * If the job is running, `GetOutput(...)` replays all previous output if any, from the job before streaming new messages. The stream is closed when the job reaches a final state
+    * If called when the process is in a final state, `GetOutput(...)` sends all job output and closes the stream.
+    
+## Client/CLI
+The cli is the interface for the user to send commands to the server. Assuming the name of the cli executable is teleport, then the command format is: 
+```shell
+> teleport [OPTIONS] COMMAND [ARG...]
+ ```
+The full set of commands will be provided through the help message which can be invoked with:
+```shell
+> teleport --help
+```
```

Comment:
please list the available commands and how the options get passed in
this can be just an example end-to-end flow of starting a job, checking status and output and stopping it

Reply:
All available commands with an example will be provided in the updated design doc